File: client/src/app/_models/userParams.ts
import { User } from "./user";

/**
 * Acts as the filters when getting members
 */
export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    PageSize = 5;
    orderBy = 'lastActive';

    constructor(user: User | null) {
        this.gender = user?.gender === 'female' ? 'male' : 'female';
    }
}

File: API/Helpers/UserParams.cs
using System;

namespace API.Helpers;

public class UserParams : PaginationParams
{
    // these are for filtering
    public string? Gender { get; set; }
    public string? CurrentUsername { get; set; }
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 100;
    public string OrderBy { get; set; } = "lastActive";
}
